src/foundry/foundry.js/clientDocuments/combat.d.ts

```diff
@@ -16,7 +18,202 @@ declare global {
+    nextRound(): Promise<this>;
+
+    /** Advance the combat to the next turn */
+    nextTurn(): Promise<this>;
+
+    /** @override */
+    prepareDerivedData(): void;
+
+    /** Rewind the combat to the previous round */
+    previousRound(): Promise<this>;
+
+    /** Rewind the combat to the previous turn */
+    previousTurn(): Promise<this>;
+
+    /** Reset all combatant initiative scores, setting the turn back to zero */
+    resetAll(): Promise<this>;
+
+    /**
+     * Roll initiative for one or multiple Combatants within the Combat entity
+     * @param ids - A Combatant id or Array of ids for which to roll
+     * @param options - Additional options which modify how initiative rolls are created or presented.
+     *                  default `{}`
+     * @returns  A promise which resolves to the updated Combat entity once updates are complete.
+     */
+    rollInitiative(ids: string | string[], options?: InitiativeOptions): Promise<this>;
+
+    /**
+     * Roll initiative for all combatants which have not already rolled
+     * @param options - Additional options forwarded to the Combat.rollInitiative method
+     *                  default `{}`
+     */
+    rollAll(options?: InitiativeOptions): this['rollInitiative'];
+
+    /**
+     * Roll initiative for all non-player actors who have not already rolled
+     * @param options - Additional options forwarded to the Combat.rollInitiative method
+     *                  default `{}`
+     */
+    rollNPC(options?: InitiativeOptions): this['rollInitiative'];
+
+    /**
+     * Assign initiative for a single Combatant within the Combat encounter.
+     * Update the Combat turn order to maintain the same combatant as the current turn.
+     * @param id - The combatant ID for which to set initiative
+     * @param value - A specific initiative value to set
+     */
+    setInitiative(id: string, value: number): Promise<void>;
+
+    /** Return the Array of combatants sorted into initiative order, breaking ties alphabetically by name. */
+    setupTurns(): InstanceType<ConfiguredDocumentClass<typeof Combatant>>[];
+
+    /** Begin the combat encounter, advancing to round 1 and turn 1 */
+    startCombat(): Promise<this>;
+
+    /**
+     * Define how the array of Combatants is sorted in the displayed list of the tracker.
+     * This method can be overridden by a system or module which needs to display combatants in an alternative order.
+     * By default sort by initiative, next falling back to name, lastly tie-breaking by combatant id.
+     */
+    protected _sortCombatants(
+      a: InstanceType<ConfiguredDocumentClass<typeof Combatant>>,
+      b: InstanceType<ConfiguredDocumentClass<typeof Combatant>>
+    ): number;
+
+    protected _onCreate(data: this['data']['_source'], options: DocumentModificationOptions, userId: string): void;
+
+    protected _onUpdate(
+      changed: DeepPartial<this['data']['_source']> & Record<string, unknown>,
```

Comment:
```suggestion
      changed: DeepPartial<this['data']['_source']>,
```

(I know it's like you typed in other places, too, but it's actually incorrect, I checked... We'll have to clean this up at some point...)